app/src/Containers/EventSearchContainer.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react'
+import styled from 'styled-components/native'
+import { ScrollView, SafeAreaView, Button, Text } from 'react-native'
+import TopOfApp from '@/Components/TopOfApp'
+import underDevelopmentAlert from '@/Utils/UnderDevelopmentAlert'
+import DateTimePicker from '@react-native-community/datetimepicker'
+import FreeSearchBar from '@/Components/FreeSearchBar';
+
+const Heading = styled.Text`
+  font-weight: bold;
+  font-size: 18px;
+  margin: 15px 15px 0px 15px;
+`
+const Label = styled.Text`
+  font-weight: bold;
+  font-size: 14px;
+  margin: 15px 15px 0px 15px;
+`
+const SectionView = styled.View`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+`
+const EventButton = styled.TouchableOpacity`
+  width: 28%;
+  height: 50px;
+  margin: 20px 0px 0px 15px;
+  padding: 5px;
+  background-color: #e3e3e3;
+  border: ${props => `1px solid ${props.theme.colors.staBlack}`};
+  display: flex;
+  justify-content: center;
+`
+const EventTitle = styled.Text`
+  display: flex;
+  text-align: center;
+`
+
+const EventSearchContainer = () => {
+  const PopularSearches = [
+    'Talk to the hiring manager',
+    'Atlassian huddle',
+    'Orientation',
+    'Showcase',
+  ]
+  const ProjectSearches = [
+    'Sole drop-in',
+    'Volunteering app',
+    'Eleos drop-in',
+    'Climate change app',
+  ]
+  const [date, setDate] = useState(new Date()); // initial date value (today)
+  const [show, setShow] = useState(false);      // hide modal date picker
+
+  const handleSearch = (input: React.SetStateAction<string>) => {
+    console.log(input)
+  }
+  const handleSubmit = () => {
+    console.log('Submit')
+  }
+
+  const onChange = (selectedDate: any) => {
+    const currentDate = selectedDate;
+    setShow(false);
+    setDate(currentDate);
+  };
+
+  const showDatepicker = () => {
+    setShow(true);
+  };
+
+  return (
+    <SafeAreaView>
+      <ScrollView>
+        <TopOfApp />
+        <FreeSearchBar
+          handleSearch={handleSearch}
+          handleSubmit={handleSubmit}
+        />
+        <SectionView>
+          <EventButton
+            onPress={underDevelopmentAlert}>
+            <EventTitle>Today</EventTitle>
+          </EventButton>
+          <EventButton
+            onPress={underDevelopmentAlert}>
+            <EventTitle>This week</EventTitle>
+          </EventButton>
+          <EventButton
+            onPress={underDevelopmentAlert}>
+            <EventTitle>This month</EventTitle>
+          </EventButton>
+        </SectionView>
+        <EventButton
+          onPress={showDatepicker}>
```

Comment:
I'm finding if I click on the "Pick date" button, choose a date from the date picker, then click "Pick date" a second time, the app crashes -- the error message is "Invariant Violation: `value` prop must be an instance of Date object".  Could you have a look into that?  Sounds like one of the values is maybe of the wrong type?

Reply:
Fixed - the Date object was not the first thing returned by the datepicker control!